scripts/tf-tests/README.md

```diff
@@ -0,0 +1,129 @@
+# Manual integration tests using Testflinger
+
+The purpose of the `run-jobs.py` script is to test hardware-api client
+and server on the machines that are accessible via
+[Testflinger](https://github.com/canonical/testflinger). It allows us
+to test the project components on multiple machines automatically.
+
+## Requirements
+
+These scripts use [the testflinger
+snap](https://snapcraft.io/testflinger-cli), so make sure you have it
+installed on your system:
+
+```sh
+sudo snap install testflinger-cli
+```
+
+Also, the following files are required to be present in this
+directory:
+
+- `machines.txt`: This file lists the Canonical IDs of the machines on
+  which jobs will be run. Each Canonical ID should be on a separate
+  line, formatted as follows:
+
+  ```text
+  202101-28595
+  202012-28526
+  ...
+  ```
+
+- `tf-job.yaml`: This YAML template defines the job parameters. The
+  template should include the placeholder `$CANONICAL_ID`, which will
+  be replaced with each actual Canonical ID from `machines.txt` when
+  jobs are submitted.  You can modify the existing `tf-job.yaml` file
+  to run other commands or use a different Ubuntu distro.
+
+- `hwctl`: The client executable. You can create it by running `cargo
+  build --release` in the project root directory. You're supposed to
+  build the binary on the same Ubuntu release that is specified in
+  `tf-job.yaml`. The machines should also be of the same architecture
+  the binary is build for.
```

Comment:
Looks like you get an error finding the right version of glibc. For testing purposes only, I wonder if it would be worth mentioning how to compile statically so that you can run it on something even if you aren't sure of the version? `RUSTFLAGS='-C target-feature=+crt-static' cargo build --release --target x86_64-unknown-linux-gnu` worked for me to do this.